ubuntu-troubleshoot.sh

```diff
@@ -0,0 +1,269 @@
+#!/bin/bash
+
+# copy the the following content into troubleshoot.sh file
+# and execute chmod +x ubuntu-troubleshoot.sh
+#
+# run this script with the absolute path of mount point.
+# sudo ./troubleshoot.sh <mountpoint> <server-ip> 
+# parameter 1 => absolute path of mount point
+# parameter 2 => server ip address
+
+# To-do:
+# - Add colors to the output
+# - Add support for aznfs
+# - Capture only limited packets in tshark
+
+# Need super user access to certain commands
+if [ "$(id -u)" -ne 0 ]; then
+    echo "Run this script as root!"
+    exit 1
+fi
+
+# Check for the number of parameters to the script
+if [[ ($# != 2) || ("$1" == "") || ("$2" == "") ]]; then
+    echo "usage: $0 <absolute mount path> <server ip>"
+    exit 1
+fi
+
+randomnum=$$
+mountpoint=$1
+accname=""
+ipaddr=$2
+
+# Print current date and time
+echo "-------------------- Starting troubleshooting at $(date -u). --------------------"
+
+cat /proc/self/mountstats > /tmp/mountstats.baseline
+
+# Detect OS and Version
+__m=$(uname -m 2>/dev/null) || __m=unknown
+__s=$(uname -s 2>/dev/null) || __s=unknown
+
+case "${__m}:${__s}" in
+    "x86_64:Linux")
+        # To-do: Add support for other distros
+        distro_id=""
+        distro_id=$(grep "^ID=" /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+        distro_id=$(echo "$distro_id" | tr '[:upper:]' '[:lower:]')
+        if [[ $distro_id != "ubuntu" ]]; then
+            echo "[FATAL] Unsupported distro: $distro_id."
+            exit 1
+        fi
+        ;;
+    *)
+        echo "[FATAL] Unsupported platform: ${__m}:${__s}."
+        exit 1
+        ;;
+esac
+
+# Log the OS and version
+echo "Platform: ${__m}:${__s}"
+echo "Distro: $distro_id"
+
+# set -eu
+
+# Setup
+dpkg -s httping tcptraceroute ioping tshark net-tools &> /dev/null
+if [[ $? -ne 0 ]]; then
+    confirmation="N"
+    echo -n "Installing tools such as tcping, httping, tcptraceroute, ioping, and tshark for troubleshooting. "
+    echo -n "Will capture traces on port 2048 and 111 to your account endpoint. "
+    echo -n "Enter Y/y if you want to continue: "
+    read confirmation
+    if [[ $confirmation == "Y" || $confirmation == "y" ]]; then
+        wget http://www.vdberg.org/~richard/tcpping -O /usr/bin/tcping &> /dev/null
```

Comment:
let's perform all downloads at the beginning so that we can duly warn the user in case he doens't have internet connectivity and even suggest him to make the file available in some place.